modules/fscloud/main.tf

```diff
@@ -45,7 +50,7 @@ locals {
       cross_region_location         = config.cross_region_location
       storage_class                 = config.storage_class
       region_location               = config.region_location
-      resource_instance_id          = module.cos_instance.cos_instance_id
+      resource_instance_id          = module.cos_instance[0].cos_instance_id
```

Comment:
I believe a local would be required solution to here, so that the COS bucket can be created in the existing COS instance.

Assuming the previous comment about locals is chosen, it we local.cos_instance_id